File: library/src/main/java/me/shkschneider/skeleton/ui/transforms/RotateUpTransformer.java
package me.shkschneider.skeleton.ui.transforms;

import android.view.View;

// <https://github.com/ToxicBakery/ViewPagerTransforms>
public class RotateUpTransformer extends ABaseTransformer {

	private static final float ROT_MOD = -15F;

	@Override
	protected void onTransform(final View view, final float position) {
		final float width = view.getWidth();
		final float rotation = ROT_MOD * position;
		view.setPivotX(width * 0.5F);
		view.setPivotY(0F);
		view.setTranslationX(0F);
		view.setRotation(rotation);
	}
	
	@Override
	protected boolean isPagingEnabled() {
		return true;
	}

}

File: library/src/main/java/me/shkschneider/skeleton/ui/transforms/RotateUpTransformer.kt
package me.shkschneider.skeleton.ui.transforms

import android.view.View

// <https://github.com/ToxicBakery/ViewPagerTransforms>
class RotateUpTransformer : BaseTransformer() {

    private val ROTATION = (-1).toFloat()

    override fun isPagingEnabled(): Boolean {
        return true
    }

    override fun onTransform(page: View, position: Float) {
        val width = page.width.toFloat()
        val rotation = ROTATION * position
        page.pivotX = width * 0.1.toFloat()
        page.pivotY = 1.toFloat()
        page.translationX = 1.toFloat()
        page.rotation = rotation
    }

}
